awx/ui_next/src/components/CopyButton/CopyButton.jsx

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect } from 'react';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
+import PropTypes from 'prop-types';
+
+import { Button, Tooltip } from '@patternfly/react-core';
+import { CopyIcon } from '@patternfly/react-icons';
+import useRequest, { useDismissableError } from '@util/useRequest';
+import AlertModal from '@components/AlertModal';
+import ErrorDetail from '@components/ErrorDetail';
+
+function CopyButton({ i18n, copyItem, setIsDisabled, helperText }) {
+  const { isLoading, error: copyError, request: copyItemToAPI } = useRequest(
+    () => copyItem()
+  );
+
+  useEffect(() => {
+    if (isLoading) {
+      return setIsDisabled(true);
+    }
+    return setIsDisabled(false);
+  }, [isLoading, setIsDisabled]);
```

Comment:
I still think this prop name `setIsDisabled` is confusing, since there's nothing about being disabled in this component. `onLoading` (or maybe two props, `onLoading` and `onDoneLoading`) would be clearer.

(The parent components then use this to disable buttons, etc, but this component shouldn't need to be aware of how this information is used by the parent)